### PR TITLE
yoav - Bus Status (using ctre)

### PR DESCRIPTION
**Pull request by YHerm**
require ctre logger to work

### CODE REVIEW

File: src/main/java/frc/utils/CTREUtils/CanivoreStatus.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    private static final double MAX_TRANSMIT_ERRORS = 3;
+    private static final String LOG_PATH = "Bus/";
+
+    public static void logAllBusStatuses() {
+        updateBusStatus(Phoenix6Constants.CANBUS_NAME);
+        updateBusStatus(Phoenix6Constants.CANIVORE_NAME);
+    }
+
+    private static void updateBusStatus(String name){
+        var busStatus = CANBus.getStatus(name);
```
Comment (YoniKiriaty) [inline src/main/java/frc/utils/CTREUtils/CanivoreStatus.java:21]: bro no var

File: src/main/java/frc/utils/CTREUtils/CanivoreStatus.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        Logger.recordOutput(LOG_PATH + name + "/Utilization", busStatus.BusUtilization);
+        Logger.recordOutput(LOG_PATH + name + "/TimesDisconnected", busStatus.BusOffCount);
+        Logger.recordOutput(LOG_PATH + name + "/Full", busStatus.TxFullCount);
+        Logger.recordOutput(LOG_PATH + name + "/ReceiveErrorCount", busStatus.REC);
+        Logger.recordOutput(LOG_PATH + name + "/TransmitErrorCount", busStatus.TEC);
+    }
+
+    private static void reportBusAlerts(CANBus.CANBusStatus busStatus, String name) {
+        if (!busStatus.Status.isOK()) {
+            Logger.recordOutput(LogPathsConstants.ALERT_LOG_PATH + LOG_PATH + name + "/Status", busStatus.Status.getName());
```
Comment (YoniKiriaty) [inline src/main/java/frc/utils/CTREUtils/CanivoreStatus.java:37]: save this `LogPathsConstants.ALERT_LOG_PATH + LOG_PATH + name` as a variable

File: src/main/java/frc/utils/cycletimeutils/CycleTimeUtils.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+public class CycleTimeUtils {
+
+    private static double lastTime = 0;
+    private static double currentTime = 0;
+
+
+    public static void updateCycleTime() {
+        lastTime = currentTime;
+        currentTime = Timer.getFPGATimestamp();
```
Comment (YHerm) [inline src/main/java/frc/utils/cycletimeutils/CycleTimeUtils.java:14]: think about using HalUtil

File: src/main/java/frc/robot/Robot.java [hunk 4]
```diff
@@ -51,8 +52,9 @@ public void teleopInit() {
 
     @Override
     public void robotPeriodic() {
-        RoborioUtils.updateRioUtils(); // Better to be first
+        CycleTimeUtils.updateCycleTime(); // Better to be first
         CommandScheduler.getInstance().run();
```
Comment (YHerm) [inline src/main/java/frc/robot/Robot.java:56]: think about adding in cycle time util something that measure "CommandScheduler.getInstance().run();" time

**Review by YoniKiriaty — APPROVED** [review]

**Pull request merged**